docs/puml/data_model.puml

```diff
@@ -0,0 +1,85 @@
+@startuml
+!theme plain
+
+enum Season
+
+enum DayType
+
+enum ApplianceType
+
+class SwitchOnProbabilityKey {
+    Season: season
+    DayType: day_type
+    int: quarter_hour_of_day
+
+    + extract_from_datetime(datetime: instance_of_interest): SwitchOnProbabilityKey
+}
+SwitchOnProbabilityKey *-- DayType
+SwitchOnProbabilityKey *-- Season
+
+class SwitchOnProbabilities {
+    dict[SwitchOnProbabilityKey, dict[int, float]]: seasonal_probability
+
+    + __init__(): SwitchOnProbabilities
+    + get(SwitchOnProbabilityKey key): float
+
+}
+note left of SwitchOnProbabilities: "Interface" definition:\nWe receive a "matrix" of probabilities,\nwhich maps from quarter hour of the\nyear to actual switch on probability\n(accounting for so called usage probability)
+SwitchOnProbabilities *-- SwitchOnProbabilityKey
+
+abstract class ApplianceData {
+    + ApplianceType: type
+    - SwitchOnProbabilities: switch_on_probability
+
+    + __init__(\n\tSwitchOnProbabilities: type_switch_on_probability): Appliance
+    + get_switch_on_probability(datetime: instance_of_interest): float
+}
+note right of ApplianceData: Implicit assumption:\nGeneration _only_ for one\nwhole year to be calculated
+ApplianceData*-- SwitchOnProbabilities
+ApplianceData*-- ApplianceType
+
+class ApplianceDataLoadProfile extends ApplianceData{
+    TimeSeries: time_series
+
+    + __init__(\n\tfloat: usage_probability,\n\tSwitchOnProbabilities: type_switch_on_probability,\n\tTimeSeries: time_series\n\t): ApplianceDataLoadProfile
+    + get_time_series_length():
+}
+note right of TimeSeries: Mapping from quarter hour\nfrom start of operation to\nconsumed power
+
+class ApplianceDataConstantPower extends ApplianceData{
+    float: p_rated
+
+    + __init__(\n\tfloat: usage_probability,\n\tSwitchOnProbabilities: type_switch_on_probability,\n\tfloat: p_rated\n\t): ApplianceDataConstantPower
+}
+
+class Appliance {
+    - List[OperationInterval]: operation_intervals
+    - ApplianceData: appliance_data
+    + handle_simulation_step():
+    - is_turned_on(): Boolean
+    - sample_switch_on(time: datetime): Unit
+    - add_operation_interval(start: datetime): Unit

```

Comment:
Some of the methods in this UML diagram aren't implemented, right? Do you intend to do so? Otherwise, we should consider providing a draft version of the document?

Reply:
This was mainly ideation for how this might work. will deal with that in #31 added them as comment to the issue and deleted them from the uml diagram for now since they might change.